packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,345 @@
+     * @default - undefined
+     */
+    readonly containerProps?: IJobDefinitionContainer;
+
+    /**
+     * An object with various properties specific to multi-node parallel jobs.
+     *
+     * @default - undefined
+     */
+    readonly nodeProps?: IMultiNodeProps;
+
+    /**
+     * When you submit a job, you can specify parameters that should replace the
+     * placeholders or override the default job definition parameters. Parameters
+     * in job submission requests take precedence over the defaults in a job definition.
+     * This allows you to use the same job definition for multiple jobs that use the same
+     * format, and programmatically change values in the command at submission time
+     *
+     * @link https://docs.aws.amazon.com/batch/latest/userguide/job_definition_parameters.html
+     * @default - undefined
+     */
+    readonly parameters?: { [key: string]: string };
+
+    /**
+     * The number of times to move a job to the RUNNABLE status. You may specify between 1 and
+     * 10 attempts. If the value of attempts is greater than one, the job is retried on failure
+     * the same number of attempts as the value.
+     *
+     * @default 1
+     */
+    readonly retryAttempts?: number;
+
+    /**
+     * The timeout configuration for jobs that are submitted with this job definition. You can specify
+     * a timeout duration after which AWS Batch terminates your jobs if they have not finished.
+     *
+     * @default - undefined
+     */
+    readonly timeout?: Duration;
+}
+
+/**
+ * Properties for specifying multi-node properties for compute resources
+ */
+export interface IMultiNodeProps {
+    /**
+     * Specifies the node index for the main node of a multi-node parallel job. This node index value must be fewer than the number of nodes.
+     */
+    mainNode: number;
+
+    /**
+     * A list of node ranges and their properties associated with a multi-node parallel job.
+     */
+    rangeProps: INodeRangeProps[];
+
+    /**
+     * The number of nodes associated with a multi-node parallel job.
+     */
+    count: number;
+}
+
+/**
+ * Properties for a multi-node batch job
+ */
+export interface INodeRangeProps {
+    /**
+     * The container details for the node range.
+     */
+    container: IJobDefinitionContainer;
+    /**
+     * The minimum node index value to apply this container definition against. If omitted, the value zero is used.
+     * You may nest node ranges, for example 0:10 and 4:5, in which case the 4:5 range properties override the 0:10 properties.
+     */
+    fromNodeIndex?: number;
+    /**
+     * The maximum node index value to apply this container definition against. If omitted, the highest value is used relative
+     * to the number of nodes associated with the job. You may nest node ranges, for example 0:10 and 4:5,
+     * in which case the 4:5 range properties override the 0:10 properties.
+     */
+    toNodeIndex?: number;
+}
+
+/**
+ * Properties of a job definition
```

Comment:
`An interface representing a job definition - either a new one, created with the CDK, using the {@link JobDefinition} class, or existing ones, referenced using the {@link JobDefinition.fromJobDefinitionArn} method.`.

Reply:
Thank you taking the time to write this up. This is perfect 👍